File: python-code/day18/exercises.py
import re
from collections import Counter

paragraph = '''I love teaching.
If you do not love teaching what else can you love.
I love Python if you do not love something which can give you all the capabilities to develop an application what else can you love.'''
txt = re.sub('[.|\n]', ' ', paragraph)
word_dict = {}
word_list = txt.split()
for word in word_list:
    if word in word_dict:
        word_dict[word] += 1
    else:
        word_dict[word] = 1
count = Counter(word_dict)
print(count.most_common()[:1])

points = ['-1', '2', '-4', '-3', '-1', '0', '4', '8']
sorted_points = sorted(map(int, points))
distance = sorted_points[len(sorted_points)-1] - sorted_points[0]
print('distance = ' + str(distance))

value = input('Please input a value:')
if not(value[0].isalpha() or value[0] == '_'):
    print('False')
else:
    for i in value[1:]:
        if not(i.isalnum() or i == '_'):
            print('False')
            break
    else:
        print('True')

sentence = '''%I $am@% a %tea@cher%, &and& I lo%#ve %tea@ching%;.
There $is nothing; &as& mo@re rewarding as educa@ting &and& @emp%o@wering peo@ple.
;I found tea@ching m%o@re interesting tha@n any other %jo@bs. %Do@es thi%s mo@tivate yo@u to be a tea@cher!?'''
clean_text = re.sub('[%$@&,#;?!]', '', sentence)
print(clean_text)
txt = re.sub('[.|\n]', ' ', clean_text)
word_dict = {}
word_list = txt.split()
for word in word_list:
    if word in word_dict:
        word_dict[word] += 1
    else:
        word_dict[word] = 1
count = Counter(word_dict)
print(count.most_common()[:3])

File: javascript/30_days_of_js/day12/exercises.js
// @ts-check
Object.defineProperty(exports, "__esModule", {
  value: true
});

const text = 'He earns 4000 euro from salary per month, 10000 euro annual bonus, 5500 euro online courses per month.';
const nums = text.match(/\d+/g);
console.log('The result is', (parseInt(nums[0]) + parseInt(nums[2])) * 12 + parseInt(nums[1]));

const description = 'The position of some particles on the horizontal x-axis -1, 2, -4, -3 and -1 in the negative direction, 0 at origin, 4 and 8 in the positive direction.';
const points = description.match(/[^]-?[\d+$]/g);
const sortedPoints = points.map(point => parseInt(point)).sort((a, b) => a - b);
console.log('distance: ', sortedPoints[sortedPoints.length - 1] - sortedPoints[0]);


let pattern = /^[^a-zA-Z_$]|[^0-9a-zA-Z_$]", "_"/g;
const isValidVariable = (str) => {
  if (str.match(pattern) != null) {
    return false;
  } else {
    return true;
  }
}
console.log(isValidVariable('firstName'));
console.log(isValidVariable('_1$'));
console.log(isValidVariable('1first'));
console.log(isValidVariable('@!'));

const paragraph = `I love teaching. If you do not love teaching what else can you love. I love Python if you do not love something which can give you all the capabilities to develop an application what else can you love.`;

const tenMostFreauentWords = (str) => {
  const res = [];
  const words = str.match(/\w+/g);
  const wordSet = new Set(words);
  for (const word of wordSet) {
    let count = 0;
    for (const element of words) {
      if (element === word) count++;
    }
    res.push({
      word: word,
      count: count
    });
  }
  return res.sort((a, b) => b.count - a.count).slice(0, 10);
};

console.log(tenMostFreauentWords(paragraph));

const sentence = `%I $am@% a %tea@cher%, &and& I lo%#ve %tea@ching%;. There $is nothing; &as& mo@re rewarding as educa@ting &and& @emp%o@wering peo@ple. ;I found tea@ching m%o@re interesting tha@n any other %jo@bs. %Do@es thi%s mo@tivate yo@u to be a tea@cher!?`;

let sentenceAfter = sentence.replace(/[^a-zA-Z\d\s.]/g, '');
console.log(tenMostFreauentWords(sentenceAfter).slice(0, 7));